subsys/bluetooth/host/classic/avdtp.c

```diff
@@ -340,602 +339,594 @@ static struct bt_avdtp_sep *avdtp_get_sep(uint8_t stream_endpoint_id)
+	}
+	k_work_cancel_delayable(&session->timeout_work);
+	if (msg_type == BT_AVDTP_ACCEPT) {
+		bt_avdtp_set_state_lock(CTRL_REQ(req)->sep, AVDTP_STREAMING);
+	} else if (msg_type == BT_AVDTP_REJECT) {
+		avdtp_handle_reject(buf, req);
+	}
+
+	if (req->status == BT_AVDTP_SUCCESS) {
 		avdtp_set_status(req, buf, msg_type);
-		bt_avdtp_clear_req(session);
-		if (req->func != NULL) {
-			req->func(req);
-		}
+	}
+	bt_avdtp_clear_req(session);
+	if (req->func != NULL) {
+		req->func(req);
 	}
 }
 
-static void avdtp_close_handler(struct bt_avdtp *session, struct net_buf *buf, uint8_t msg_type,
-				uint8_t tid)
+static void avdtp_close_cmd(struct bt_avdtp *session, struct net_buf *buf, uint8_t tid)
 {
-	if (msg_type == BT_AVDTP_CMD) {
-		int err = 0;
-		int ret;
-		struct bt_avdtp_sep *sep;
-		struct net_buf *rsp_buf;
-		uint8_t error_code = 0;
+	int err = 0;
+	int ret;
+	struct bt_avdtp_sep *sep;
+	struct net_buf *rsp_buf;
+	uint8_t error_code = 0;
 
-		sep = avdtp_get_cmd_sep(buf);
-		avdtp_sep_lock(sep);
-		if ((sep == NULL) || (session->ops->close_ind == NULL)) {
+	sep = avdtp_get_cmd_sep(buf, &error_code);
+	avdtp_sep_lock(sep);
+	if ((sep == NULL) || (session->ops->close_ind == NULL)) {
+		err = -ENOTSUP;
+	} else {
+		if (!(sep->state & (AVDTP_OPEN | AVDTP_STREAMING))) {
 			err = -ENOTSUP;
+			error_code = BT_AVDTP_BAD_STATE;
 		} else {
-			if (!(sep->state & (AVDTP_OPEN | AVDTP_STREAMING))) {
-				err = -ENOTSUP;
-				error_code = BT_AVDTP_BAD_STATE;
-			} else {
-				err = session->ops->close_ind(session, sep, &error_code);
-			}
-		}
-
-		rsp_buf = avdtp_create_reply_pdu(err ? BT_AVDTP_REJECT : BT_AVDTP_ACCEPT,
-						 BT_AVDTP_PACKET_TYPE_SINGLE, BT_AVDTP_CLOSE, tid);
-		if (!rsp_buf) {
-			avdtp_sep_unlock(sep);
-			return;
+			err = session->ops->close_ind(session, sep, &error_code);
 		}
+	}
 
-		if (err) {
-			if (error_code == 0) {
-				error_code = BT_AVDTP_BAD_ACP_SEID;
-			}
-			LOG_DBG("close err code:%d", error_code);
-			net_buf_add_u8(rsp_buf, error_code);
-		} else {
-			bt_avdtp_set_state(sep, AVDTP_CLOSING);
-		}
+	rsp_buf = avdtp_create_reply_pdu(err ? BT_AVDTP_REJECT : BT_AVDTP_ACCEPT,
+					 BT_AVDTP_PACKET_TYPE_SINGLE, BT_AVDTP_CLOSE, tid);
+	if (!rsp_buf) {
+		avdtp_sep_unlock(sep);
+		return;
+	}
 
-		ret = bt_l2cap_chan_send(&session->br_chan.chan, rsp_buf);
-		if (ret) {
-			net_buf_unref(rsp_buf);
-			LOG_ERR("Error:L2CAP send fail - result = %d", ret);
+	if (err) {
+		if (error_code == 0) {
+			error_code = BT_AVDTP_BAD_ACP_SEID;
 		}
-		if (!err && !ret) {
-			bt_avdtp_set_state(sep, AVDTP_IDLE);
-		}
-		avdtp_sep_unlock(sep);
+		LOG_DBG("close err code:%d", error_code);
+		net_buf_add_u8(rsp_buf, error_code);
 	} else {
-		struct bt_avdtp_req *req = session->req;
+		bt_avdtp_set_state(sep, AVDTP_CLOSING);
+	}
 
-		if (req == NULL) {
+	ret = bt_l2cap_chan_send(&session->br_chan.chan, rsp_buf);
+	if (ret) {
+		net_buf_unref(rsp_buf);
+		LOG_ERR("Error:L2CAP send fail - result = %d", ret);
+	}
+	if (!err && !ret) {
```

Comment:
`else if (!err) {` perhaps? You should also fix the variable naming & usage here, since it's super confusing to have both `err` and `ret`. Is it possible to remove one of them? If not, you need to use names which are more descriptive.